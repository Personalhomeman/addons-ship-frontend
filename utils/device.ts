import * as ReactDeviceDetect from 'react-device-detect';
import * as CompareVersions from 'compare-versions';

export const isAndroid = () => ReactDeviceDetect.isAndroid;
export const isIOS = () => ReactDeviceDetect.isIOS;
export const osVersion = () => ReactDeviceDetect.osVersion;
export const mobileModel = () => ReactDeviceDetect.mobileModel;
export const compareVersions = (firstVersion: string, secondVersion: string) => CompareVersions(firstVersion, secondVersion)
