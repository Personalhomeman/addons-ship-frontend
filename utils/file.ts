import fetch from 'node-fetch';

export const uploadFileToS3 = async (file: File | undefined, presignedUrl: string | undefined) => {
  if (!file || !presignedUrl) throw new Error('File or presignedUrl is invalid');

  const options = {
    method: 'put',
    body: file as any
  };

  return fetch(presignedUrl, options).then(resp => {
    if (!resp.ok) throw new Error(`${resp.status}: ${resp.statusText}`);
  });
};
