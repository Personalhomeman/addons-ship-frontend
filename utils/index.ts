import mapKeys from 'lodash/mapKeys';
import mapValues from 'lodash/mapValues';
import camelCase from 'lodash/camelCase';
import snakeCase from 'lodash/snakeCase';
import isPlainObject from 'lodash/isPlainObject';

export const actionTypeCreator = (prefix: string) => (actionType: TemplateStringsArray) => `${prefix}_${actionType}`;

export const camelizeKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => camelCase(key)) as unknown) as T;

export const camelizeKeysDeep = <T>(input: Object): T =>
  mapValues(camelizeKeys(input), (value: any) => (isPlainObject(value) ? camelizeKeysDeep(value) : value)) as T;

export const snakifyKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => snakeCase(key)) as unknown) as T;

export const snakifyKeysDeep = <T>(input: Object): T =>
  mapValues(snakifyKeys(input), (value: any) => (isPlainObject(value) ? snakifyKeysDeep(value) : value)) as T;
