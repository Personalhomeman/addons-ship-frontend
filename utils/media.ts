import { useState, useEffect } from 'react';

type Handler = (e: MediaQueryListEvent) => void;

const getListeners = (queries: string[], handler: Handler) =>
  queries.map(query => {
    const listener = window.matchMedia(query);

    listener.addListener(handler);

    return listener;
  });

export const mediaQuery = (...rules: string[]) => {
  const mediaRules = rules.map(r => `(min-width: ${r})`);

  const stateHandlers = rules.map(() => useState(false));

  const getStateHandler = (media: string) => {
    const idx = mediaRules.indexOf(media);

    return stateHandlers[idx];
  };

  useEffect(() => {
    const handler = ({ matches, media }: MediaQueryListEvent) => {
      const [, set] = getStateHandler(media);

      set(matches);
    };

    const listeners = getListeners(mediaRules, handler);

    listeners.forEach(({ media, matches }) => {
      const [, set] = getStateHandler(media);
      set(matches);
    });

    return () => listeners.forEach(l => l.removeListener(handler));
  });

  return stateHandlers.map(([isMatch]) => isMatch);
};
