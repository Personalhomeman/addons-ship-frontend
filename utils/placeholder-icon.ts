import { placeholderIcons } from '@/config';
import { ProjectType } from '@/models/app';

export const getIcon = (appIcon?: string, projectType?: ProjectType) => {
  let icon = appIcon;

  if (!icon) {
    icon = placeholderIcons[projectType!] || placeholderIcons.other;
  }

  const suffix = Math.random()
    .toString(16)
    .replace(/^0\./, '');

  return `${icon}?${suffix}`;
};
