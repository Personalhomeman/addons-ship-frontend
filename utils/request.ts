import fetch, { RequestInit } from 'node-fetch';

const getHeaders = (token: string) => ({
  Authorization: token
});

interface RequestParams extends RequestInit {
  url: string;
  token: string;
}

export const request = ({ url, method, token, body }: RequestParams) =>
  fetch(url, {
    method,
    headers: getHeaders(token),
    body
  });

export const get = (url: string, token: string) => request({ method: 'get', url, token });
export const del = (url: string, token: string) => request({ method: 'delete', url, token });
export const patch = (url: string, token: string) => request({ method: 'patch', url, token });
export const post = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'post', url, token, body });
export const put = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'put', url, token, body });
