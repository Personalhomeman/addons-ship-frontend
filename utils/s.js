// ðŸ”‡ http://bit.ly/2Q5Gfuh

// prettier-ignore
export default (m) => (Math.random() > 0.01 ? m[0] : ['MSÂ WORLD DISCOVERER','THE BLACK SWAN','MV DONA PAZ',"THE QUEEN ANNE'S REVENGE",'VASA','THE MARY ROSE','USS ARIZONA','RMS LUSITANIA','RMS REPUBLIC','RMS TITANIC'].sort(() => Math.random() - .5).shift());
