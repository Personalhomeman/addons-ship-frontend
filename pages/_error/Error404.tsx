import React from 'react';
import { Button, Buttons, Flex, Status404, Text, Icon } from '@bitrise/bitkit';

import { mediaQuery } from '@/utils/media';

type Props = {
  appSlug?: string;
};

export default ({ appSlug }: Props) => {
  const [isDesktop] = mediaQuery('60rem');

  return (
    <Flex direction="vertical" grow alignChildrenVertical="middle" paddingVertical="x10">
      <Status404 />

      {appSlug && (
        <Flex direction="horizontal" alignChildren="middle" style={{ marginTop: -100 }}>
          <Button
            Component="a"
            href={`https://app.bitrise.io/app/${appSlug}/addon/addons-ship/login_page`}
            level="splash"
            size="large"
            gap="x2"
          >
            <Icon name="Ship" />
            <Text>Reauthenticate</Text>
          </Button>
        </Flex>
      )}

      <Text align="middle" config="5" margin="x8">
        Do you prefer the deep unkown, or looking for something else?
      </Text>

      <Buttons
        alignChildren={isDesktop ? 'middle' : undefined}
        fullWidth={!isDesktop}
        gap={isDesktop ? 'x6' : 'x3'}
        paddingHorizontal="x10"
      >
        <Button Component="a" href="https://app.bitrise.io/dashboard/apps" level="light" size="small">
          Bitrise Dashboard
        </Button>

        <Button
          Component="a"
          href="https://devcenter.bitrise.io/tips-and-tricks/tips-and-tricks-index/"
          level="light"
          size="small"
        >
          Learn Bitrise Tricks
        </Button>

        <Button Component="a" href="https://careers.bitrise.io/" level="light" size="small">
          Check Open Positions
        </Button>
      </Buttons>
    </Flex>
  );
};
