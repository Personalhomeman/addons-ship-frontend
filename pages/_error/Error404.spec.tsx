jest.mock('@/utils/media');

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { mediaQuery } from '@/utils/media';

import Error404 from './Error404';

describe('Error404', () => {
  beforeEach(() => {
    (mediaQuery as jest.Mock).mockReturnValue([true]);
  });

  describe('renders ok', () => {
    test('on desktop', () => {
      const tree = shallowToJson(shallow(<Error404 />));
      expect(tree).toMatchSnapshot();
    });

    test('on mobile', () => {
      (mediaQuery as jest.Mock).mockReturnValueOnce([false]);

      const tree = shallowToJson(shallow(<Error404 />));
      expect(tree).toMatchSnapshot();
    });
  });

  test('with app slug', () => {
    const tree = shallowToJson(shallow(<Error404 appSlug="an-app-slug" />));
    expect(tree).toMatchSnapshot();
  });
});
