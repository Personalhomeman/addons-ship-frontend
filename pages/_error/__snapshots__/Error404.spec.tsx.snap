// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 renders ok on desktop 1`] = `
<Flex
  alignChildrenVertical="middle"
  direction="vertical"
  grow={true}
  paddingVertical="x10"
>
  <Status404 />
  <Text
    align="middle"
    config="5"
    margin="x8"
  >
    Do you prefer the deep unkown, or looking for something else?
  </Text>
  <Buttons
    alignChildren="middle"
    fullWidth={false}
    gap="x6"
    paddingHorizontal="x10"
  >
    <Button
      Component="a"
      href="https://app.bitrise.io/dashboard/apps"
      level="light"
      size="small"
    >
      Bitrise Dashboard
    </Button>
    <Button
      Component="a"
      href="https://devcenter.bitrise.io/tips-and-tricks/tips-and-tricks-index/"
      level="light"
      size="small"
    >
      Learn Bitrise Tricks
    </Button>
    <Button
      Component="a"
      href="https://careers.bitrise.io/"
      level="light"
      size="small"
    >
      Check Open Positions
    </Button>
  </Buttons>
</Flex>
`;

exports[`Error404 renders ok on mobile 1`] = `
<Flex
  alignChildrenVertical="middle"
  direction="vertical"
  grow={true}
  paddingVertical="x10"
>
  <Status404 />
  <Text
    align="middle"
    config="5"
    margin="x8"
  >
    Do you prefer the deep unkown, or looking for something else?
  </Text>
  <Buttons
    fullWidth={true}
    gap="x3"
    paddingHorizontal="x10"
  >
    <Button
      Component="a"
      href="https://app.bitrise.io/dashboard/apps"
      level="light"
      size="small"
    >
      Bitrise Dashboard
    </Button>
    <Button
      Component="a"
      href="https://devcenter.bitrise.io/tips-and-tricks/tips-and-tricks-index/"
      level="light"
      size="small"
    >
      Learn Bitrise Tricks
    </Button>
    <Button
      Component="a"
      href="https://careers.bitrise.io/"
      level="light"
      size="small"
    >
      Check Open Positions
    </Button>
  </Buttons>
</Flex>
`;

exports[`Error404 with app slug 1`] = `
<Flex
  alignChildrenVertical="middle"
  direction="vertical"
  grow={true}
  paddingVertical="x10"
>
  <Status404 />
  <Flex
    alignChildren="middle"
    direction="horizontal"
    style={
      Object {
        "marginTop": -100,
      }
    }
  >
    <Button
      Component="a"
      gap="x2"
      href="https://app.bitrise.io/app/an-app-slug/addon/addons-ship/login_page"
      level="splash"
      size="large"
    >
      <Icon
        name="Ship"
        size="1.5rem"
      />
      <Text>
        Reauthenticate
      </Text>
    </Button>
  </Flex>
  <Text
    align="middle"
    config="5"
    margin="x8"
  >
    Do you prefer the deep unkown, or looking for something else?
  </Text>
  <Buttons
    alignChildren="middle"
    fullWidth={false}
    gap="x6"
    paddingHorizontal="x10"
  >
    <Button
      Component="a"
      href="https://app.bitrise.io/dashboard/apps"
      level="light"
      size="small"
    >
      Bitrise Dashboard
    </Button>
    <Button
      Component="a"
      href="https://devcenter.bitrise.io/tips-and-tricks/tips-and-tricks-index/"
      level="light"
      size="small"
    >
      Learn Bitrise Tricks
    </Button>
    <Button
      Component="a"
      href="https://careers.bitrise.io/"
      level="light"
      size="small"
    >
      Check Open Positions
    </Button>
  </Buttons>
</Flex>
`;
